ruby/ql/lib/codeql/ruby/security/internal/CleartextSources.qll

```diff
@@ -208,36 +218,42 @@ module CleartextSources {
     override string describe() { result = "a write to " + name }
   }
 
-  /** An assignment that may assign a password to a variable */
-  private class AssignPasswordVariableSource extends Source {
+  /** An assignment that may assign sensitive data to a variable */
+  private class AssignSensitiveVariableSource extends Source {
     string name;
 
-    AssignPasswordVariableSource() {
-      // avoid safe values assigned to presumably unsafe names
-      not this instanceof NonCleartextPassword and
-      name.regexpMatch(maybePassword()) and
-      not nameIsNotSensitive(name) and
-      exists(Assignment a |
-        this.asExpr().getExpr() = a.getRightOperand() and
-        a.getLeftOperand().getAVariable().getName() = name
+    AssignSensitiveVariableSource() {
+      exists(SensitiveDataClassification classification |
+        // avoid safe values assigned to presumably unsafe names
+        not this instanceof NonCleartextSensitive and
+        nameIndicatesSensitiveData(name, classification) and
+        not classification = SensitiveDataClassification::id() and
```

Comment:
Ditto the above comment, so perhaps we could move this check out into a helper predicate (`isRelevantClassification(SensitiveDataClassification classification)` or similar).